extensions/flyway/deployment/src/main/java/io/quarkus/flyway/ScannerTransformer.java

```diff
@@ -0,0 +1,116 @@
+package io.quarkus.flyway;
+
+import static org.objectweb.asm.Opcodes.ALOAD;
+import static org.objectweb.asm.Opcodes.ASTORE;
+import static org.objectweb.asm.Opcodes.DUP;
+import static org.objectweb.asm.Opcodes.GETFIELD;
+import static org.objectweb.asm.Opcodes.INVOKEINTERFACE;
+import static org.objectweb.asm.Opcodes.INVOKESPECIAL;
+import static org.objectweb.asm.Opcodes.NEW;
+import static org.objectweb.asm.Opcodes.POP;
+import static org.objectweb.asm.Opcodes.PUTFIELD;
+import static org.objectweb.asm.Opcodes.RETURN;
+
+import java.util.function.BiFunction;
+
+import org.flywaydb.core.internal.scanner.Scanner;
+import org.flywaydb.core.internal.scanner.classpath.ResourceAndClassScanner;
+import org.objectweb.asm.ClassVisitor;
+import org.objectweb.asm.MethodVisitor;
+
+import io.quarkus.flyway.runtime.QuarkusPathLocationScanner;
+import io.quarkus.gizmo.Gizmo;
+
+class ScannerTransformer implements BiFunction<String, ClassVisitor, ClassVisitor> {
```

Comment:
Can we add a link to the issue we created in Flyway with just a note explaining why need this class. This way we can easily track the issue once it is merged, we can get rid of this class. WDYT?

Reply:
I was thinking that the extensive commit message would be enough, no?

Reply:
Yes, it helps to have it there too in the commit message. I think having it in the class will make quick when someone stumble through the class in the future to just copy the issue and check if it fixed upstream or especially during reviews of Flyway's auto upgrades by the bot.

Reply:
Well, since I'm the one that opened the issue, I'll obviously keep an eye for updates to it.

But I'll add the comment just in case

Reply:
Otherwise, I had the same concern as @Sanne but after following your discussions on this PR I suppose we can live with this for a while till the issue is fixed hence my suggestion for tracking it explicitly in the code.

Reply:
> Well, since I'm the one that opened the issue, I'll obviously keep an eye for updates to it.
> 
> But I'll add the comment just in case

Nice, thanks.

Reply:
Added

Reply:
Great. All good for me.